automated install/basic-install.sh

```diff
@@ -188,7 +188,7 @@ os_check() {
         local remote_os_domain valid_os valid_version valid_response detected_os detected_version display_warning cmdResult digReturnCode response
         remote_os_domain="versions.pi-hole.net"
 
-        detected_os=$(grep "\bID\b" /etc/os-release | cut -d '=' -f2 | tr -d '"')
+        detected_os=$(grep '^ID=' /etc/os-release | cut -d '=' -f2 | tr -d '"')
```

Comment:
This line is causing the test to fail.

```
[gw0] [ 86%] FAILED test_automated_install.py::test_os_check_fails 
```

https://github.com/pi-hole/pi-hole/blob/b5e0f142ccefe9b1b59b94d7e82d3052efbfdcd0/test/test_automated_install.py#L565-L582

Reply:
Because the test creates an invalid `os-release` file with leading spaces, which `/etc/os-release` does not have. Two possible fixes:
```sh
     cat <<EOT > /etc/os-release 
ID=UnsupportedOS 
VERSION_ID="2" 
EOT
```
```sh
echo -e 'ID=UnsupportedOS\nVERSION_ID="2"' > /etc/os-release
```
Not sure which `echo` is invoked? `bash` and `/bin/echo` require the `-e` flag to interpret `\n` as newline, `dash` (bourne shell internal echo) must have it omitted, as it prints the `-e` literally.

Reply:
Specifically:
```
E       AssertionError: assert 'Unsupported OS detected: UnsupportedOS' in '  [\x1b[1;31m✗\x1b[0m] \x1b[1;31mUnsupported OS detected:  2\x1b[0m\n      If you are seeing this message and you do ...Help\x1b[0m category:\n      \x1b[1;32mhttps://discourse.pi-hole.net/c/bugs-problems-issues/community-help/\x1b[0m\n\n'
E        +  where '  [\x1b[1;31m✗\x1b[0m] \x1b[1;31mUnsupported OS detected:  2\x1b[0m\n      If you are seeing this message and you do ...Help\x1b[0m category:\n      \x1b[1;32mhttps://discourse.pi-hole.net/c/bugs-problems-issues/community-help/\x1b[0m\n\n' = CommandResult(command=b't\n    source /opt/pihole/basic-install.sh\n    os_check\n    ', exit_status=1, stdout=b'  [\x...se.pi-hole.net/c/bugs-problems-issues/community-help/\x1b[0m\n\n', stderr=b'/bin/bash: line 1: t: command not found\n'

```

Which is interesting, because it should result in the same thing.

![image](https://user-images.githubusercontent.com/1998970/143306210-45ead7fc-f9ef-4809-96d8-2def7c752e98.png)

Reply:
As said, the code is fine with real `/etc/os-release` but the test creates an invalid `/etc/os-release` and hence needs a fix: https://github.com/pi-hole/pi-hole/pull/4088#discussion_r756389473

Reply:
@ChillerDragon 
Could you please rebase onto current `development` branch? The tests should now all go through.

Reply:
Rebase did not work too well. Sorry for the merge commit.

Reply:
Maintainers can squash and merge the PR it the multiple commits were an issue 🙂. All tests did now pass 👍.